src/main/java/com/stripe/model/Charge.java

```diff
@@ -493,4 +493,104 @@ public Charge update(Map<String, Object> params, RequestOptions options) throws
         options);
   }
   // </editor-fold>
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class AlternateStatementDescriptors extends StripeObject {
+    String kana;
+    String kanji;
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class FraudDetails extends StripeObject {
+    public static final String USER_REPORT = "user_report";
+
+    String userReport;
+    String stripeReport;
+  }
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public static class Level3 extends StripeObject {
+    String customerReference;
+    List<LineItem> lineItems;
+    String merchantReference;
+    String shippingAddressZip;
+    String shippingFromZip;
+    Long shippingAmount;
+
+    @Getter
+    @Setter
+    @EqualsAndHashCode(callSuper = false)
+    public static class LineItem extends StripeObject {
```

Comment:
I wonder if we might end up with classes way too deep because of this and if this hurts readability. Just flagging while reviewing.

Reply:
Yeah, that's my main concern with this approach, and is why I left `LegalEntity` as a standalone class.